core/internal/cltest/mocks.go

```diff
@@ -392,6 +392,7 @@ func NewLegacyChainsWithMockChain(t testing.TB, ethClient evmclient.Client, cfg
 	scopedCfg := evmtest.NewChainScopedConfig(t, cfg)
 	ch.On("ID").Return(scopedCfg.EVM().ChainID())
 	ch.On("Config").Return(scopedCfg)
+	ch.On("HeadTracker").Return(nil)
```

Comment:
don't like this, but NewSimulatedHeadTracker returns a conflicted type the mock chain expects.. Open to suggestions

Reply:
I think this is ok for now since no one needs to mock the headtracker for the tests that use this. Otherwise, if any test wants to mock the headtracker in the future, they can update this method to pass an instance of it like this but don't think you have to worry about that change.
`func NewLegacyChainsWithMockChain(t testing.TB, ethClient evmclient.Client, cfg legacyevm.AppConfig, ht httypes.HeadTracker)`